papermill/tests/test_translators.py

```diff
@@ -187,6 +187,73 @@ def test_translate_codify_scala(parameters, expected):
     assert translators.ScalaTranslator.codify(parameters) == expected
 
 
+# C# section
+@pytest.mark.parametrize(
+    "test_input,expected",
+    [
+        ("foo", '"foo"'),
+        ('{"foo": "bar"}', '"{\\"foo\\": \\"bar\\"}"'),
+        ({"foo": "bar"},
+            'new Dictionary<Object,Object?>(new [] '
+            + '{ new KeyValuePair<Object,Object>("foo" , "bar") })'),
+        ({"foo": '"bar"'},
+            'new Dictionary<Object,Object?>(new [] '
+            + '{ new KeyValuePair<Object,Object>("foo" , "\\"bar\\"") })'),
+        (["foo"], 'new [] { "foo" }'),
+        (["foo", '"bar"'], 'new [] { "foo", "\\"bar\\"" }'),
+        ([{"foo": "bar"}],
+            'new [] { new Dictionary<Object,Object?>('
+            + 'new [] { new KeyValuePair<Object,Object>("foo" , "bar") }) }'),
+        (12345, '12345'),
+        (-54321, '-54321'),
+        (1.2345, '1.2345'),
+        (-5432.1, '-5432.1'),
+        (2147483648, '2147483648L'),
+        (-2147483649, '-2147483649L'),
+        (True, 'true'),
+        (False, 'false')
+    ],
+)
+def test_translate_type_csharp(test_input, expected):
+    assert translators.CSharpTranslator.translate(test_input) == expected
+
+
+@pytest.mark.parametrize(
+    "test_input,expected",
+    [("", '//'), ("foo", '// foo'), ("['best effort']", "// ['best effort']")],
+)
+def test_translate_comment_csharp(test_input, expected):
+    assert translators.CSharpTranslator.comment(test_input) == expected
+
+
+@pytest.mark.parametrize(
+    "input_name,input_value,expected",
+    [
+        ("foo", '""', 'var foo = "";'),
+        ("foo", '"bar"', 'var foo = "bar";'),
+    ],
+)
+def test_translate_assign_csharp(input_name, input_value, expected):
+    assert translators.CSharpTranslator.assign(input_name, input_value) == expected
+
+
+@pytest.mark.parametrize(
+    "parameters,expected",
+    [
+        ({"foo": "bar"}, '// Parameters\nvar foo = "bar";\n'),
+        ({"foo": True}, '// Parameters\nvar foo = true;\n'),
+        ({"foo": 5}, '// Parameters\nvar foo = 5;\n'),
+        ({"foo": 1.1}, '// Parameters\nvar foo = 1.1;\n'),
+        ({"foo": ['bar', 'baz']}, '// Parameters\nvar foo = new [] { "bar", "baz" };\n'),
+        ({"foo": {'bar': 'baz'}},
+            '// Parameters\nvar foo = new Dictionary<Object,Object?>(new [] '
```

Comment:
I struggle to get this to compile. Is there a specific version of C# this works (can't test the .net kernel specifically from this computer atm)?

Seems like `var foo = new Dictionary<Object,Object?> { { "bar", "baz" } }\n` would work (and be a little easier to read)?

Reply:
Do we need `using System.Collections.Generic;` for this to work in the kernel? We could inject that into the parameter cell if it is needed.

Reply:
... still working on this ...

Reply:
Re. `using`: YAGNI, apparently. I confirmed that it works without explicitly bringing that in.

Reply:
Ok, great -- just figured I'd check for the kernel. The setup must be providing some basic things on behalf of the user.